cpp/src/join/join_common_utils.cuh

```diff
@@ -101,27 +102,23 @@ class row_is_valid {
  *
  * @tparam Comparator The row comparator type to perform row equality comparison from row indices.
  */
-template <typename Comparator = row_equality>
+template <typename DeviceComparator>
 class pair_equality {
  public:
-  pair_equality(table_device_view lhs,
-                table_device_view rhs,
-                nullate::DYNAMIC has_nulls,
-                null_equality nulls_are_equal = null_equality::EQUAL)
-    : _check_row_equality{has_nulls, lhs, rhs, nulls_are_equal}
-  {
-  }
-
-  pair_equality(Comparator const d_eqcomp) : _check_row_equality{std::move(d_eqcomp)} {}
+  pair_equality(DeviceComparator check_row_equality) : _check_row_equality{check_row_equality} {}
 
   template <typename LhsPair, typename RhsPair>
   __device__ __forceinline__ bool operator()(LhsPair const& lhs, RhsPair const& rhs) const noexcept
   {
-    return lhs.first == rhs.first and _check_row_equality(rhs.second, lhs.second);
+    using experimental::row::lhs_index_type;
+    using experimental::row::rhs_index_type;
+
+    return lhs.first == rhs.first and
+           _check_row_equality(lhs_index_type{rhs.second}, rhs_index_type{lhs.second});
```

Comment:
Are you sure with the order? Why mixing lhs and rhs here? Such as `lhs_index_type{rhs.second}`?

Reply:
`lhs_index_type` is a wrapper to strongly type order of arguments since the new lex comparator supports both `(lhs, rhs)` and `(rhs, lhs)`. `lhs_index_type{rhs.second}` is there because the comparator is constructed as `(probe, build)`

Reply:
I was confused because why not `lhs_index_type{lhs.}` instead of `lhs_index_type{rhs.}`? Does `rhs.second` is the index of the left table?